multiwallet.go

```diff
@@ -37,23 +37,32 @@ type MultiWallet struct {
 }
 
 func NewMultiWallet(rootDir, dbDriver, netType string) (*MultiWallet, error) {
-	rootDir = filepath.Join(rootDir, netType)
-	initLogRotator(filepath.Join(rootDir, logFileName))
 	errors.Separator = ":: "
 
-	chainParams := utils.ChainParams(netType)
-	if chainParams == nil {
-		return nil, errors.E("unsupported network type: %s", netType)
+	chainParams, err := utils.ChainParams(netType)
+	if err != nil {
+		return nil, err
+	}
+
+	rootDir = filepath.Join(rootDir, netType)
+	err = os.MkdirAll(rootDir, 0700)
+	if err != nil {
+		return nil, errors.Errorf("failed to create rootDir: %v", err)
+	}
+
+	err = initLogRotator(filepath.Join(rootDir, logFileName))
+	if err != nil {
+		return nil, errors.Errorf("failed to init logRotator: %v", err.Error())
 	}
 
 	walletsDb, err := storm.Open(filepath.Join(rootDir, walletsDbName))
 	if err != nil {
 		log.Errorf("Error opening wallets database: %s", err.Error())
 		if err == bolt.ErrTimeout {
 			// timeout error occurs if storm fails to acquire a lock on the database file
-			return nil, errors.E("wallets database is in use by another process")
+			return nil, errors.E(ErrWalletDatabaseInUse)
```

Comment:
+1 for this. Client apps can now check the error returned against this constant and easily recognize and handle issues where a wallet db is already opened. As mentioned earlier, don't think similar constant stuff is needed for the log rotator errors.

Reply:
No no no. Why are you using strings? What exactly is the point of returning an error type if all branches return a string? Either use an error type or use a string.

Reply:
The error contains an exported string variable so it can be referenced from mobile to determine what error was thrown. It's returning an error type to trigger the exception in a try-catch block(for mobile) and gomobile would ignore a function with 2 return types.

> Any function type all of whose parameters and results have
>   supported types. Functions must return either no results,
>   one result, or two results where the type of the second is
>   the built-in 'error' type.

https://godoc.org/golang.org/x/mobile/cmd/gobind#hdr-Type_restrictions.

Reply:
Apologies for the confusion, I meant to return an error with a string literal rather than a string constant in the log rotator function. Of course, it is atypical to return a string instead of an error.

In any case, I've also clarified my comments on the log rotator error constants.